Atmos 2.0

KAKTYC900 opened the pull request:
## О ПР`е
Добавлено 4 новых вида газов: Дейтерий, Гипер-Ноблий, Анти-Ноблий и BZ. Последние три производятся в новой структуре - Фазовом генераторе частиц. Фазовый генератор частиц способен перерабатывать газы, параллельно вырабатывая небольшое количество электроэнергии.

Тип: feat
Изменения: Добавлены 4 новых газа и многоблочная структура - Фазовый генератор частиц. В нём топливом выступает Дейтерий и в нём же производятся остальные газы. Побочная выработка электроэнергии зависит от реакции, происходящей в контурах генератора. Гипер-Ноблий улучшает физические качества живых существ, но наносит клеточные повреждения при вдыхании. Анти-Ноблий останавливает любые реакции в атмосфере, если его содержание превышает 5 молей. BZ немного излечивает молей от клеточного урона и используется ими для дыхания, а также он усыпляет слаймолюдов. Дейтерий поджигает дышащих им существ. Сам же Фазовый генератор частиц находится в Т1 исследовании ветки промышленности отдела РНД (Атмосферные технологии).

<img width="505" height="822" alt="изображение" src="https://github.com/user-attachments/assets/b935c76e-c00e-4332-9426-6c24422fe34b" />

## Технические детали
Нет

## Изменения кода официальных разработчиков
AtmosphereSystem.Hotspot.cs, PortableScrubberComponent.cs, AtmosphereSystem.Gases.cs, GasReactionPrototype.cs,  Atmospherics.cs, SharedVentScrubberComponent.cs, NodeGroupID.cs, spray-painter.ftl, sensor.yml, vox.yml, canister_groups.yml, engineering.yml, industrial.yml, Gasses.xml, meta.json



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

# Примечания к выпуску

* **Новые функции**
  * Добавлены новые газы: Дейтерий, Гипер-Ноблий, Анти-Ноблий и BZ с соответствующими свойствами и визуализацией.
  * Добавлена новая система фазового генератора частиц для производства энергии с использованием газов.
  * Добавлены новые контейнеры для хранения газов, включая канистры и резервуары различных типов.
  * Расширен набор фильтруемых газов в системах вентиляции и газоочистки.

* **Документация**
  * Добавлены локализационные строки на русском языке для всех новых газов и оборудования.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

KAKTYC900 committed:
Atmos Update

Добавление 4 новых газов и Фазового генератора частиц.

The-Fiafi commented on Content.Server/Atmos/Portable/PortableScrubberComponent.cs:
```diff
@@ -33,7 +33,11 @@ public sealed partial class PortableScrubberComponent : Component
-            Gas.Hydrogen /// Imperial Added Hydrogen
+            Gas.Hydrogen, /// Imperial Added Hydrogen
+            Gas.Deuterium, /// Imperial Atmos Update start
+            Gas.HyperNoblium,
```
Комментарии

KAKTYC900 replied:
Что именно с ними не так?

The-Fiafi commented on Content.Server/Imperial/Atmos/PPGenerator/Components/PPGeneratorComponent.cs:
```diff
@@ -0,0 +1,36 @@
+public sealed partial class PPGeneratorComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("rampFactor")]
```
Передача строки излишняя. Оно автоматически генерируется из названия поля

KAKTYC900 committed:
Фиксы

Вынесение 5f в константу, указываемую в Atmospherics.cs. добавление стейта assembly

KAKTYC900 committed:
Delete left.png

Удаление left.png

KAKTYC900 committed:
Small fixes

Исправление GasPhazeReactionPrototype.cs и добавление файла лицензии.

KAKTYC900 committed:
Fix

Фиксы локализации и PPGeneratorSystem.cs

KAKTYC900 committed:
Fix

Фиксы прототипов реакции, локализации и мелкие изменения.

The-Fiafi commented on Content.Server/Atmos/Reactions/FrezonCoolantReaction.cs:
```diff
@@ -13,6 +13,9 @@ public sealed partial class FrezonCoolantReaction : IGasReactionEffect
 {
     public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
     {
+        var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

KAKTYC900 replied:
Исправлено.

The-Fiafi commented on Content.Server/Atmos/Reactions/FrezonProductionReaction.cs:
```diff
@@ -14,6 +14,9 @@ public sealed partial class FrezonProductionReaction : IGasReactionEffect
 {
     public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
     {
+        var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

The-Fiafi commented on Content.Server/Atmos/Reactions/N2ODecompositionReaction.cs:
```diff
@@ -13,6 +13,9 @@ public sealed partial class N2ODecompositionReaction : IGasReactionEffect
 {
     public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
     {
+        var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

The-Fiafi commented on Content.Server/Atmos/Reactions/PlasmaFireReaction.cs:
```diff
@@ -11,6 +11,9 @@ public sealed partial class PlasmaFireReaction : IGasReactionEffect
     {
         public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
         {
+            var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

The-Fiafi commented on Content.Server/Atmos/Reactions/TritiumFireReaction.cs:
```diff
@@ -11,6 +11,9 @@ public sealed partial class TritiumFireReaction : IGasReactionEffect
     {
         public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
         {
+            var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

The-Fiafi commented on Content.Server/Atmos/Reactions/AmmoniaOxygenReaction.cs:
```diff
@@ -10,6 +10,9 @@ public sealed partial class AmmoniaOxygenReaction : IGasReactionEffect
 {
     public ReactionResult React(GasMixture mixture, IGasMixtureHolder? holder, AtmosphereSystem atmosphereSystem, float heatScale)
     {
+        var initialAntiNoblium = mixture.GetMoles(Gas.AntiNoblium); // Imperial Atmos Update start
```
Создайте интерфейс условия для реакции и добавьте его как поле в GasReactionPrototype.
В методе React сделайте итерацию по условиям.

The-Fiafi commented on Content.Server/Imperial/Atmos/PPGenerator/Systems/PPGeneratorSystem.cs:
```diff
@@ -0,0 +1,244 @@
+    [Dependency] private readonly IPrototypeManager _protoMan = default!;
+
+    private EntityQuery<NodeContainerComponent> _nodeContainerQuery;
+    private GasPhazeReactionPrototype[] _gasReactions = Array.Empty<GasPhazeReactionPrototype>();
```
Зачем? Вы используете это поле в одном месте и постоянно его оверрайдите

KAKTYC900 replied:
Исправлено.

KAKTYC900 committed:
Fix

Изменение PPGeneratorSystem.cs, GasReactionPrototype.cs и AtmosphereSystem.Gases.cs
Изменение кода реакций.

KAKTYC900 committed:
Update AtmosphereSystem.Gases.cs

Мелкий фикс комментария
